modules/apps/item-selector/item-selector-taglib/src/main/resources/META-INF/resources/repository_entry_browser/js/ItemSelectorRepositoryEntryBrowserManagementToolbarPropsTransformer.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {openSelectionModal} from 'frontend-js-web';
+
+export default function propsTransformer({portletNamespace, ...otherProps}) {
+	const openCreateAIImage = (aiCreatorDALLEEnabled, aiImageCreatorURL) => {
+		if (!aiCreatorDALLEEnabled) {
+			Liferay.componentReady(
+				`${portletNamespace}ItemSelectorRepositoryEntryBrowserConfigureAIModal`
+			).then((configureAIModal) => {
+				configureAIModal.open();
+			});
+		}
+		else {
+			openSelectionModal({
+				size: 'lg',
+				title: Liferay.Language.get('create-ai-image'),
+				url: aiImageCreatorURL,
+			});
+		}
+	};
+
+	return {
+		...otherProps,
+		onCreationMenuItemClick: (event, {item}) => {
+			if (item?.data?.action === 'openAICreateImage') {
+				openCreateAIImage(
```

Comment:
As a rule, when there is only one action, we inline the function

Reply:
Done.

Please see https://github.com/liferay-lima/liferay-portal/pull/4919/commits/367c6e43e3af565ce7c4a6113d6e3de2a5cd7a4f